### ISSUE TITLE
Allow transition to ended if all tracks except subtitle track have ended

### ISSUE THREAD

**Issue by BROUDING** [google/ExoPlayer#5896]
EXOPLAYER_VERSION=2.10.0

I'm testing with a video and 2 subtitles, and 1 subtitle's timeline is 3seconds longer than a video.

What happens now is, video is finished, but I don't get `COMPLETED` event until subtitle finish playing... and also "current time" keep updates and "video play time" didn't updated.

What I want to do is, I want to focus on Video play time and get the `COMPLETED` event when video is finished.... what should I do?

Thank you !!!

**Comment by BROUDING** [comment]
Confirmed that `durationUs` in `createMediaSource` doesn't affect the run time of any medias..

**Comment by ojw28** [comment]
ExoPlayer considers the end of the media to be the point at which all media has been played, including subtitles. _Why_ is the subtitle longer than the video? It sounds like the media is kind of broken to me.

I don't think we have any plans to provide an easy mechanism for working around this.

**Comment by tonihei** [comment]
You can also try to use a `ClippingMediaSource` to clip the subtitle source to the length of the video. This requires to know that in advance though.

**Comment by BROUDING** [comment]
@ojw28 @tonihei 
First, I appreciate your reply and time.
It is human error for subtitle makers, but it would have be awesome if media finish playing when video is done playing no matter what, since human error can happen anytime.

Now I'll try to work on `ClippingMediaSource` and see if the subtitle actually clipped as the `duration` I manually set...

**Comment by BROUDING** [comment]
@tonihei 
`ClippingMediaSource` doesn't work as expected in my use case...

I approached like this >> I got a DASH media source with `DashMediaSource.Factory` and 3 different subtitles by `ClippingMediaSource(Factory.createTextSampleFormat(source), durationUs)` and made them to a single media source via `MergingMediaSource`. And the user can change the subtitle manually with custom subtitle choice view. 

Media plays well so as subtitle, all I need now is just clipping subtitle length.. (since MergedMedia doesn't focus on Video timeline but on longest MediaSource...)

And I've set `durationUs` as `3000L or 3000000L` for testing, and it didn't work as I expected...

Any idea why...?

**Comment by BROUDING** [comment]
I just confirmed that ClippingMediaSource works on `Videos`, but not on `Subtitles`...
Follow up >> [issue#5317](https://github.com/google/ExoPlayer/issues/5317)

@ojw28 I guess it's still a bug.. 
btw, player in iOS seems focusing on video timeline than other media sources. My college has no problem without any changes... 

Feel free to close this issue if there's no updates !

**Comment by tonihei** [comment]
Sorry, I forgot about #5317. Clipping text and metadata is not supported yet and implementation is tracked by the linked issue.

**Comment by tonihei** [comment]
Looks like you just found that out yourself :)

**Comment by BROUDING** [comment]
Nah, I should have searched more- before I raise this issue, but hopped that this would be solved by now- hahah

But...now my issue is that I cannot get event when the video is finished playing........until subtitle makers don't make mistake...

**Comment by ojw28** [comment]
Changed this to a feature request. I think it should probably be an option, possibly passed via the track `Format`.

**Comment by icbaker** [comment]
`ClippingMediaSource` **does** now work on subtitles (#5317). Given that allows 'broken' media with different length tracks to be 'fixed' by truncating all the tracks to the same length, I'm not sure we also need a text-specific option to allow playback to finish 'early' - so I suggest we close this 'won't fix: intended behavior'.
